plugins/utils/lookup_apps.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+"""
+Lookup the "apps" directory to generate apps configuration
+"""
+import os
+import re
+
+from ansible.plugins.filter.core import to_nice_yaml
+
+PROJECT_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), "../.."))
+APPS_DIRECTORY = os.path.join(PROJECT_ROOT, "apps")
+IGNORE_DIRS_REGEX = "^_.*$"
+APP_VOLUMES_DIR = "_volumes"
+SERVICE_CONFIG_DIR = "_configs"
+
+
+def lookup_apps(apps_path):
+    """
+    Lookup ready-to-deploy applications following recommended Arnold applications'
+    tree, e.g. for the "richie" application and the "apps" ``apps_path``:
+
+        apps/richie
+        ├── templates
+        │   ├── elasticsearch
+        │   │   ├── dc.yml.j2
+        │   │   └── svc.yml.j2
+        │   ├── nginx
+        │   │   ├── _configs
+        │   │   │   └── richie.conf.j2
+        │   │   ├── dc.yml.j2
+        │   │   └── svc.yml.j2
+        │   ├── postgresql
+        │   │   ├── dc.yml.j2
+        │   │   ├── ep.yml.j2
+        │   │   └── svc.yml.j2
+        │   ├── richie
```

Comment:
Could be called "app" to be generic?

Reply:
I'm not sure since the service name can be "injected" in its templates.

Reply:
You already have the name "richie" from the parent directory... can you get it from there?  It would be nice if the structure inside an app could be fixed.

Reply:
Richie is the name of the app but it's also the name of the main service in this app.
For the moment it's not a problem IMHO.
In the future I think whe can work on a app with only one main service (like an Ansible role) in the dir and a dependencies file (like the [meta/main.yml](https://docs.ansible.com/ansible/latest/reference_appendices/galaxy.html?highlight=meta#dependencies) in role dir).